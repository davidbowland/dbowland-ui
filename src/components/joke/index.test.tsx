import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import Joke from './index'
import JokeService from '@services/jokes'

jest.mock('@services/jokes')

describe('Joke component', () => {
  const jokes = [{ joke: 'Ha' }, { joke: 'Ha!' }]
  const getRandomJokes = jest.fn().mockResolvedValue(jokes)

  beforeAll(() => {
    const mockMath = Object.create(global.Math)
    mockMath.random = () => 0
    global.Math = mockMath

    JokeService.getRandomJokes = getRandomJokes
  })

  test('Ensure node snapshot match', () => {
    const { container } = render(<Joke />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Ensure snapshot match', async () => {
    const { container, findByText } = render(<Joke initialize={true} />)
    await findByText('Next joke')
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Ensure button works to change joke displayed', async () => {
    const { container, findByText } = render(<Joke initialize={true} />)
    const nextJokeButton: HTMLButtonElement = await findByText('Next joke') as HTMLButtonElement
    nextJokeButton.click()
    expect(container.firstChild).toMatchSnapshot()
  })
})
