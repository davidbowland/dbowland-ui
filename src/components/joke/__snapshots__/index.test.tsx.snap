// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke component Ensure button works to change joke displayed 1`] = `
<h1>
  Ha!
</h1>
`;

exports[`Joke component Ensure node snapshot match 1`] = `<h1 />`;

exports[`Joke component Ensure snapshot match 1`] = `
<h1>
  Ha
</h1>
`;
